src/link/abstractlink.h

```diff
@@ -212,10 +212,24 @@ class AbstractLink : public QObject
     void write(const char* data, int size)
     {
         if(size > 0) {
-            emit sendData(QByteArray(data, size));
-        };
+            if (_type == LinkType::Serial){
```

Comment:
I put this case here because I have not tested other link types. I think that maybe we should favor the `writeSync` logic over the `emit sendData()` logic for everyone.